### PR TITLE
Creator viewer profile

### PR DESCRIPTION
**Pull request by GuilaumeBrassard**

### CODE REVIEW

File: cryptus-app/components/profile/creator_profile/creator_profile.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React, { useState, useEffect } from "react";
+import s from "./creator_profile.module.scss";
+
+import CreatorProfileInfos from "./creator_profile_infos/creator_profile_infos";
+import CreatorHeader from "./creator_header/creator_header";
+//import ProfileWalletViewer from "../../wallet_viewer/profile_wallet_viewer/profile_wallet_viewer"
+//import Footer from "../../footer/footer"
+
+export default function CreatorProfile(props) {
```
Comment (turcottep) [inline cryptus-app/components/profile/creator_profile/creator_profile.tsx]: define type of props:
props:{variable1: string, variable2: number, etc}

File: cryptus-app/components/profile/creator_profile/creator_profile_infos/creator_profile_infos.module.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
+@use "/styles/variables" as v;
+
+.container {
+  display: flex;
+  flex-direction: column;
+
+  width: 100%;
+  height: min(150px, 250px); //For too long description
```
Comment (turcottep) [inline cryptus-app/components/profile/creator_profile/creator_profile_infos/creator_profile_infos.module.scss]: this always returns 150px, not sure of the goal

File: cryptus-app/components/profile/creator_profile/creator_profile_infos/creator_profile_infos.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React, { useState, useEffect } from "react";
+import s from "./creator_profile_infos.module.scss";
+
+import ViewerProfilePicture from "../../viewer_profile/viewer_profile_infos/viewer_profile_picture/viewer_profile_picture";
+import EditNftsButton from "./edit_nfts_button/edit_nfts_button";
+import EditProfileButton from "./edit_profile_button/edit_profile_button";
+import ViewerProfileName from "../../viewer_profile/viewer_profile_infos/viewer_profile_name/viewer_profile_name";
+import ViewerProfileDescription from "../../viewer_profile/viewer_profile_infos/viewer_profile_description/viewer_profile_description";
+
+export default function CreatorProfileInfos(props) {
```
Comment (turcottep) [inline cryptus-app/components/profile/creator_profile/creator_profile_infos/creator_profile_infos.tsx]: define props

File: cryptus-app/components/profile/creator_profile/creator_profile_infos/edit_profile_button/edit_profile_button.module.scss [hunk 4]
```diff
@@ -0,0 +1,24 @@
+@use "/styles/variables" as v;
+
+.container {
+  flex: 1 1 auto;
+
+  display: flex;
+  align-items: end;
+}
+
+.editbutton {
```
Comment (turcottep) [inline cryptus-app/components/profile/creator_profile/creator_profile_infos/edit_profile_button/edit_profile_button.module.scss:10]: spacing, rounding and color is not what was in the document:
image

File: cryptus-app/components/profile/profile.tsx [hunk 5]
```diff
@@ -1,93 +1,21 @@
...
-              strokeLinejoin="round"
-              strokeWidth="2"
-              d="M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z"
-            />
-          </svg>
-        </div>
-      </div>
-    );
-  }
+export default function Profile(props) {
```
Comment (turcottep) [inline cryptus-app/components/profile/profile.tsx]: define props

File: cryptus-app/components/profile/viewer_profile/viewer_profile.module.scss [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  flex: 1 1 auto;
+
+  flex-direction: column;
+  align-items: center;
+}
+
+.wallet { //Placeholder
+  height: 800px; //Just to see it scroll
+  margin-bottom: v.$footer-h;
+  width: 100%;
```
Comment (turcottep) [inline cryptus-app/components/profile/viewer_profile/viewer_profile.module.scss:13]: width is not 100% when rendered:
image

File: cryptus-app/components/profile/viewer_profile/viewer_profile.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import React, { useState, useEffect } from "react";
+import s from "./viewer_profile.module.scss";
+
+import ViewerProfileInfos from "./viewer_profile_infos/viewer_profile_infos";
+import ViewerHeader from "./viewer_header/viewer_header";
+//import ProfileWalletViewer from "../../wallet_viewer/profile_wallet_viewer/profile_wallet_viewer"
+//import Footer from "../../footer/footer"
+
+export default function ViewerProfile(props) {
```
Comment (turcottep) [inline cryptus-app/components/profile/viewer_profile/viewer_profile.tsx]: define props

File: cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_description/viewer_profile_description.tsx [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import React, { useState, useEffect } from "react";
+import s from "./viewer_profile_description.module.scss";
+
+
+export default function ViewerProfileDescription(props) {
```
Comment (turcottep) [inline cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_description/viewer_profile_description.tsx]: define props

File: cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_infos.tsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import React, { useState, useEffect } from "react";
+import s from "./viewer_profile_infos.module.scss";
+
+import ViewerProfilePicture from "./viewer_profile_picture/viewer_profile_picture";
+import ViewerProfileName from "./viewer_profile_name/viewer_profile_name";
+import ViewerProfileDescription from "./viewer_profile_description/viewer_profile_description";
+
+export default function ViewerProfileInfos(props) {
```
Comment (turcottep) [inline cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_infos.tsx]: define props

File: cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_name/viewer_profile_name.tsx [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import React, { useState, useEffect } from "react";
+import Checkmarks from "./checkmarks/checkmarks";
+import s from "./viewer_profile_name.module.scss";
+
+
+export default function ViewerProfileName(props) {
```
Comment (turcottep) [inline cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_name/viewer_profile_name.tsx]: define props

**Review by turcottep — DISMISSED** [review]

File: cryptus-app/components/profile/viewer_profile/viewer_profile.module.scss (hunk 6 above)
Comment (GuilaumeBrassard) [reply cryptus-app/components/profile/viewer_profile/viewer_profile.module.scss:13]: true j'ai changé le nom du container dans creator et pas dans viewer

File: cryptus-app/components/profile/creator_profile/creator_profile_infos/creator_profile_infos.module.scss (hunk 2 above)
Comment (GuilaumeBrassard) [reply cryptus-app/components/profile/creator_profile/creator_profile_infos/creator_profile_infos.module.scss]: Forgot to convert for 20vh

File: cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_description/viewer_profile_description.tsx [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import React, { useState, useEffect } from "react";
+import s from "./viewer_profile_description.module.scss";
+
+
+export default function ViewerProfileDescription(props: {description: String}) {
+  return (
+    <div className={s.container}>
+      {props.description}
```
Comment (turcottep) [inline cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_description/viewer_profile_description.tsx:8]: include one liner functions in parent, don't create a whole document

File: cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_name/checkmarks/checkmarks.tsx [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import React, { useState, useEffect } from "react";
+import s from "./checkmarks.module.scss";
+
+
+export default function Checkmarks() {
+  return (
+    <div className={s.container}>
+      <img className={s.checkmarks} src="./images/twitter_checkmark.png"></img>
```
Comment (turcottep) [inline cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_name/checkmarks/checkmarks.tsx]: remove one liner functions

File: cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_name/viewer_profile_name.tsx [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import React, { useState, useEffect } from "react";
+import Checkmarks from "./checkmarks/checkmarks";
+import s from "./viewer_profile_name.module.scss";
+
+
+export default function ViewerProfileName(props: {displayName: String}) {
+  return (
+    <div className={s.container}>
+      <div>{props.displayName}</div>
```
Comment (turcottep) [inline cryptus-app/components/profile/viewer_profile/viewer_profile_infos/viewer_profile_name/viewer_profile_name.tsx]: remove one liner

**Review by turcottep — CHANGES_REQUESTED** [review]

**Review by turcottep — APPROVED** [review]

**Pull request merged**